sdk/log/crate/src/logger.rs

```diff
@@ -212,10 +212,13 @@ pub trait Log {
 
 /// Implement the log trait for unsigned integer types.
 macro_rules! impl_log_for_unsigned_integer {
-    ( $type:tt, $max_digits:literal ) => {
+    ( $type:tt ) => {
         impl Log for $type {
             #[inline]
             fn write_with_args(&self, buffer: &mut [MaybeUninit<u8>], args: &[Argument]) -> usize {
+                // The maximum number of digits that the type can have.
+                const MAX_DIGITS: usize = const { $type::MAX.ilog10() as usize + 1 };
```

Comment:
nit: i don't think this makes a difference, but is the second `const` needed? I was able to do this in a test, outside of a macro:
```suggestion
                // The maximum number of digits that the type can have.
                const MAX_DIGITS: usize = $type::MAX.ilog10() as usize + 1;
```